ibis/omniscidb/client.py

```diff
@@ -546,28 +569,61 @@ def _get_table_schema(self, table_name, database=None):
             database, table_name = table_name_
         return self.get_schema(table_name, database)
 
-    def _execute(self, query, results=True):
+    def _execute_query(self, dml, **kwargs):
+        query = self.query_class(self, dml, **kwargs)
+        return query.execute(**kwargs)
+
+    def _execute(self, query, results=True, execution_type=None):
         """
+        Compile and execute Ibis expression using this backend client
+        interface, returning results in-memory in the appropriate object type
 
-        query:
-        :return:
+        Parameters
+        ----------
+        query : string
+          DML or DDL statement
+        execution_type: {1, 2, 3}, optional
+          Determine the type of the execution (default EXECUTION_TYPE_CURSOR).
+          The execution types available are:
+            1 - EXECUTION_TYPE_ICP
+            2 - EXECUTION_TYPE_ICP_GPU
+            3 - EXECUTION_TYPE_CURSOR
+        results : boolean, default False
+          Pass True if the query as a result set
+
+        Returns
+        -------
+        output : execution type dependent
+          EXECUTION_TYPE_ICP:
+            pandas.DataFrame or
+            geopandas.GeoDataFrame (if corresponds)
+          EXECUTION_TYPE_ICP_GPU: cudf.dataframe.DataFrame
+          EXECUTION_TYPE_CURSOR:
+            pandas.DataFrame or
+            geopandas.GeoDataFrame (if corresponds)
         """
         if isinstance(query, (DDL, DML)):
             query = query.compile()
 
-        if self.execution_type == EXECUTION_TYPE_ICP:
-            execute = self.con.select_ipc
-        elif self.execution_type == EXECUTION_TYPE_ICP_GPU:
-            execute = self.con.select_ipc_gpu
-        else:
-            execute = self.con.cursor().execute
+        if execution_type is None:
+            execution_type = self.execution_type
+
+        self._check_execution_type(execution_type)
 
         cursor = (
             OmniSciDBGeoCursor
             if FULL_GEO_SUPPORTED
             else OmniSciDBDefaultCursor
         )
 
+        if execution_type == EXECUTION_TYPE_ICP:
+            execute = self.con.select_ipc
+        elif execution_type == EXECUTION_TYPE_ICP_GPU:
+            execute = self.con.select_ipc_gpu
+            cursor = OmniSciDBGPUCursor
```

Comment:
Can a situation arise where `FULL_GEO_SUPPORTED` resolves to `False` but a user uses `EXECUTION_TYPE_ICP_GPU`? I assume something unintended would happen if that were the case. Should probably try to handle that case.

Reply:
for FULL_GEO_SUPPORTED and EXECUTION_TYPE_ICP_GPU probably `OmniSciDBGPUCursor` should be the desired choice ... but maybe we could add a flag like `force_geo_output` or something like that

Reply:
if execution_type is EXECUTION_TYPE_ICP_GPU and FULL_GEO_SUPPORTED is False
so 
`execute = self.con.select_ipc_gpu` and  `cursor = OmniSciDBGPUCursor`

so I think it should work normally